### DISCUSSION TITLE
Report decimal/float instead of integer

### DISCUSSION THREAD

**Discussion by nikobearrr** [netdata/netdata#14003] [Ideas]
I have a plugin which ends up updating the metrics like so:

```python
used = float(used)
avail = float(avail)
self.data[path] = avail * 100.0 / (avail + used)
```

But this is send as an integer to the chart. How can I have it with 2-3 decimals?

**Comment by thiagoftsm** [comment]
Hello @nikobearrr ,

If you want to show some decimal values instead only integers, you have to multiply the integer by a higher value and when you create the [dimension](https://learn.netdata.cloud/docs/agent/collectors/plugins.d#dimension), you need to specify the divisor. When Netdata shows the number for you, it will use the specified value and display the data like you want.

To give an example for you, please, take a look in next image:

![Screenshot_20221116_114556](https://user-images.githubusercontent.com/49162938/202172503-253863a9-05e4-43ed-93ef-9bf2416ba882.png)

as you can see I have decimal values that are created because we divide collector value by 1024 https://github.com/netdata/netdata/blob/cde5456408b4af12a56cc11979cb9dfab777ecc4/collectors/proc.plugin/proc_meminfo.c#L208-L223 .

Best regards!

**Reply by ilyam8** [reply]
@thiagoftsm is correct. @nikobearrr see [External Plugin API Data Collection](https://learn.netdata.cloud/docs/agent/collectors/plugins.d#data-collection):

> **value**

> is the collected value, only integer values are collected. If you want to push fractional values, multiply this value by 100 or 1000 and set the DIMENSION divider to 1000.

**Reply by nikobearrr** [reply]
Thank you both @thiagoftsm and @ilyam8.
That did solve the issue of showing correct decimals on my chart. However, when exporting, it is still the underlying integer. Is there no way of exporting it as decimal to a 3rd party TSDB?

**Reply by thiagoftsm** [reply]
Hello @nikobearrr ,

I am glad that first issue was fixed, now let us talk about the next.

Are you specifying `data source` when you send data? You can take a look [here](https://learn.netdata.cloud/docs/agent/exporting#operation-modes) for more details. My question was only to confirm, but I think now we have a feature request and no more a doubt. 

After your answer I will talk with our product team (ping @shyamvalsan and @sashwathn ) to plan this new feature.

Best regards!

**Reply by nikobearrr** [reply]
Yes, I am using graphite as TSDB. And I do use the `as-collected` mechanism. Looking at the options I do not think I can achieve what I am interested it, so it must be a feature request.

**Reply by thiagoftsm** [reply]
Thank you to confirm.

We had an internal talk about this and I changed now the status from `Question answer` to `Ideas`.
